hub/tests/test_permissions.py

```diff
@@ -870,3 +870,221 @@ def test_generic_data_visibility(self):
         result = res.json()
 
         self.assertIsNone(result["data"]["importedDataGeojsonPoint"])
+
+
+@override_settings(ALLOWED_HOSTS=["testserver"])
+class TestLoggedOutUserForPublicSource(Setup, TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.source.can_display_points_publicly = True
+        self.source.can_display_details_publicly = True
+        self.source.save()
+
+    def test_permissions_calculator(self):
+        permissions = models.ExternalDataSource.user_permissions(None, self.source)
+        self.assertDictEqual(
+            permissions, {"can_display_points": True, "can_display_details": True}
+        )
+
+    def test_aggregate_data_count(self):
+        """
+        Logged out users can't access the sharedDataSource graphql query
+        """
+        query = """
+            query MapReportLayerGeoJSONPoint($sourceId: ID!) {
+              sharedDataSource(pk: $sourceId) {
+                id
+                importedDataCountByRegion {
+                  gss
+                  count
+                }
+              }
+            }
+        """
+
+        res = self.client.post(
+            reverse("graphql"),
+            content_type="application/json",
+            data={"query": query, "variables": {"sourceId": str(self.source.id)}},
+        )
+        result = res.json()
+
+        self.assertIsNotNone(result.get("errors", None))
+        self.assertIsNone(result["data"])
+
+    def test_generic_data_visibility(self):
+        """
+        Logged out users can't see generic data
```

Comment:
Shouldn't this depend on the data source's public sharing perms?

Reply:
I think so, but currently this is behind an auth check, so I wasn't sure.